cyences_app_for_splunk/default/savedsearches.conf

```diff
@@ -2756,20 +2756,61 @@ display.page.search.mode = fast
-|rename UserId as User \
+| iplocation ClientIP \
+| fillnull value="" Country, Region, City \
+| eval Source = ClientIP \
+| stats values(Operation) AS Operations, values(EventData) AS EventData, values(Source) as Source, values(ItemType) as ItemType, values(ApplicationDisplayName) as ApplicationDisplayName, values(EventSource) as EventSource latest(_time) as LastTime earliest(_time) as FirstTime, values(Country) as Country, values(City) as City, values(Region) as Region by SourceFileName, UserId \
+| rename UserId as User \
 | `cs_human_readable_time_format(FirstTime)` \
 | `cs_human_readable_time_format(LastTime)` \
 | `cs_user_privilege_mapping(User)` \
 | eval cyences_severity = if(is_privileged_user=="Yes", "critical","high") \
-| table User,is_privileged_user,SourceFileName,EventSource,ItemType,Operations,ApplicationDisplayName,FirstTime,LastTime,EventData,cyences_severity \
+| table User,is_privileged_user,SourceFileName,EventSource,ItemType,Operations,ApplicationDisplayName,FirstTime,LastTime,EventData,cyences_severity,Source,Country,City,Region \
 | `cs_o365_sharing_file_with_external_user_filter`
 action.cyences_notable_event_action = 1
 action.cyences_notable_event_action.param.filter_macro_name = cs_o365_sharing_file_with_external_user_filter
 action.cyences_notable_event_action.contributing_events = `cs_o365` sourcetype="o365:management:activity" _index_earliest=-61m@m _index_latest=-1m@m Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup")
 action.cyences_send_email_action = 1
 action.cyences_notable_event_action.products = Office 365
 
+[O365 - OneDrive or SharePoint Link Accessed By External User]
+disabled = 1
+enableSched = 1
+alert.track = 1
+alert.severity = 4
+alert.suppress = 0
+counttype = number of events
+quantity = 0
+relation = greater than
+cron_schedule = 4 * * * *
+description = This alert will pinpoint when an external user access a file shared with onedrive or sharepoint. \
+\
+Data Collection - Office 365 management activity data (Splunk Add-on for Office 365). 
+dispatch.earliest_time = -4h@h
+dispatch.latest_time = +4h@h
+display.general.type = statistics
+display.page.search.tab = statistics
+display.page.search.mode = fast
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_o365` sourcetype="o365:management:activity" _index_earliest=-61m@m _index_latest=-1m@m Workload IN (OneDrive, SharePoint) Operation IN ("AnonymousLinkUsed","SharingLinkUsed") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup") \
+| search UserId=anonymous \
+| iplocation ClientIP  \
+| fillnull value="" Country, Region, City \
+| eval Source = ClientIP \
```

Comment:
Doesn't seems to require "| eval Source = ClientIP".
you can use ClientIP or if you want to rename then use: | rename ClientIP as Source